tests/integration/test_model_element_relationships.py

```diff
@@ -49,9 +70,32 @@ def test_adding_relationship_via_uses_adds_to_elements():
     sys1 = model.add_software_system(name="sys1")
     sys2 = model.add_software_system(name="sys2")
 
-    sys1.uses(sys2, "uses")
-    assert len(sys1.relationships) == 1
-    assert len(list(sys1.get_relationships())) == 1
-    assert len(model.get_relationships()) == 1
-    assert len(sys2.relationships) == 1  # Currently will fail
-    assert len(list(sys2.get_relationships())) == 1
+    r = sys1.uses(sys2, "uses")
+    assert list(sys1.relationships) == [r]
+    assert list(sys1.get_relationships()) == [r]
+    assert list(model.get_relationships()) == [r]
+    assert list(sys2.relationships) == []  # relationships only contains outbound
+    assert list(sys2.get_relationships()) == [r]
+
+
+@pytest.mark.parametrize(
+    "filename",
+    ["BigBank.json"],
+)
+def test_relationships_after_deserialisation_are_consistent(filename):
+    """Make sure that relationships are consistent between the Model and the Element after deserialisation."""
+    path = DEFINITIONS / filename
+    workspace = Workspace.load(path)
+    model = workspace.model
+    atm = model.get_software_system_with_id("9")
+    mainframe = model.get_software_system_with_id("4")
+    customer = model.get_element("1")
+
+    assert len(atm.relationships) == 1
+    assert (
+        len(list(atm.get_relationships())) == 2
+    )  # One to mainframe, one from personal banking customer
+    assert len(list(atm.get_afferent_relationships())) == 1
```

Comment:
Can't do this as the method returns a generator rather than a collection.  I had a quick look on StackOverflow but didn't see any non-ugly alternatives.  Open to any other ideas!

Reply:
Can't do this as the method returns a generator rather than a collection. I had a quick look on StackOverflow but didn't see any non-ugly alternatives. Open to any other ideas!

Reply:
Right, I didn't check the return type. That's perfectly fine then.